messagingclient/client.py

```diff
@@ -48,3 +53,87 @@ def callback_wrapper(payload):
                 # terminate on any exception so that the worker isn't hung.
                 future.cancel()
                 print(f"stopped listening: {exc}")
+
+    def consume_multiple(self, queues: Union[str, List], callback: Callable = None, max_messages: int = 1):
+        '''
+        TODO: This function can completely replace consume() (i.e. be renamed consume(), deleting the older version),
+        but for development, I left the original consume() untouched and added this as a new function instead.
+        To avoid altering any code that currently calls consume(), the existing version could eventually be deleted
+        and this version could be renamed in its place.
+        
+        If one queue is provided, it will be consumed until the process is terminated.
+        If multiple queues are provided, they will be consumed in round robin fashion.
+
+        Refs:
+        https://cloud.google.com/pubsub/docs/pull
+        '''
+        print("MessagingClient.consume_multiple()")
+        print("MessagingClient.consume_multiple() queues:", queues)
+
+        if isinstance(queues, str):
+            return self.consume(queues, callback, max_messages)
+        if len(queues) == 1:
+            return self.consume(queues[0], callback, max_messages)
+        
+        if callback is None:
+            callback = _print
+
+        def _print(payload):
+            print(payload.data)
+
+        def callback_wrapper(payload):
+            """Call user callback and send acknowledge."""
+            print("MessagingClient.consume_multiple().callback_wrapper()")
+            callback(payload)
+            payload.ack()
+
+        # TODO: Note that most aspects of the subscription path are hardcoded.
+        # It could be further generalized, but for now, we will merely cycle over subscriptions.
```

Comment:
I don't think this is true, gcp requires the path to be in this format (`projects` and `subscriptions` don't change).
Since we know we're only identifying subscriptions here using this path format, they can be hardcoded.

I don't see how this can be further realized, making this TODO comment unnecessary.

Or maybe I'm misunderstanding what you mean?

Reply:
That's fine. I wasn't sure if I was misunderstanding how it might pull from different "institutions" or something. It can pull from different topics however, as that it is how I'm currently using it. The multiple subscriptions could be on different topics, so I agree that this can be removed. I'll clean it up.